### PR TITLE
Implement Shading for OpenTelemetry Instrumentation Libraries ASP.NET Core, HTTPClient and SQL using IL Repack- 

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
* Utilized IL Repack to combine and internalize the public APIs from the OpenTelemetry Instrumentation Libraries (ASP.NET Core, HTTPClient, SQL).
* Enabled `InternalsVisibleTo` to `Azure.Monitor.OpenTelemetry.AspNetCore` to allow internal access.

### CODE REVIEW

**Comment by rajkumar-rangaraj** [comment]
Need to find a right place to create `Azure.Monitor.OpenTelemetry.Instrumentation` project. CI is not happy without readme, will create a default template readme.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Azure.Monitor.OpenTelemetry.AspNetCore.csproj [hunk 1]
```diff
@@ -10,14 +10,15 @@
 
   <ItemGroup>
     <PackageReference Include="OpenTelemetry.Extensions.Hosting" />
-    <PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" />
-    <PackageReference Include="OpenTelemetry.Instrumentation.Http" />
-    <PackageReference Include="OpenTelemetry.Instrumentation.SqlClient" />
     <PackageReference Include="OpenTelemetry.ResourceDetectors.Azure" />
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Azure.Monitor.OpenTelemetry.AspNetCore.csproj:13]: should this also include the ResourceDetectors, or are we planning to internalize this?

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Azure.Monitor.OpenTelemetry.AspNetCore.csproj:13]: I'm in favor of copying the code for this package.

**Comment by rajkumar-rangaraj** [comment]
We decided to use vendoring  https://github.com/Azure/azure-sdk-for-net/pull/38848, as ILRepack does not work well in Linux and Mac systems.

**Pull request closed without merging**